tools/star.wdl

```diff
@@ -529,7 +529,7 @@ task alignment {
         String alignEndsType = "Local"
         String alignSoftClipAtReferenceEnds = "Yes"
         String alignInsertionFlush = "None"
-        String chimOutType = "Junctions"
+        String chimOutType = "WithinBAM HardClip"
```

Comment:
I would think that soft clipping is a more reasonable choice for this as a default parameter, right? That way, no information is lost from the `SEQ` field? If Arriba needs it hardclipped, then I would recommend overriding this at the workflow level.

Reply:
`HardClip` is actually the default in STAR. This only applies to `supplemental chimeric` alignments. The primary alignment should still have the full sequence, only the supplemental alignments will be hard clipped.


For example for the following read, only the `supplementary` alignments contain hard clipping.

```
A00467:240:HCN7KDRX2:2:2275:20410:25786 2481    chr1    24927   0       107H18M1S       chr14   49586604        0       GACCACCGGCACGCCCCCC     ,:F,:F,,,F,F,F,FF,F     NH:i:7  HI:i:4  AS:i:17 nM:i:0  NM:i:0  MD:Z:18 RG:Z:c949090    SA:Z:chr14,49586609,+,19S107M,0,10;
A00467:240:HCN7KDRX2:2:2275:20410:25786 2481    chr1    195452  0       107H18M1S       chr14   49586604        0       GACCACCGGCACGCCCCCC     ,:F,:F,,,F,F,F,FF,F     NH:i:7  HI:i:3  AS:i:17 nM:i:0  NM:i:0  MD:Z:18 RG:Z:c949090    SA:Z:chr14,49586609,+,19S107M,0,10;
A00467:240:HCN7KDRX2:2:2275:20410:25786 2481    chr1    46341588        0       110H16M chr14   49586604        0       CACCGGCACGCCCCCC        ,:F,,,F,F,F,FF,F        NH:i:7  HI:i:1  AS:i:13 nM:i:1  NM:i:1  MD:Z:0G15       RG:Z:c949090    SA:Z:chr14,49586606,+,16S110M,0,10;
A00467:240:HCN7KDRX2:2:2275:20410:25786 2481    chr1    46341589        0       111H15M chr9    9442086 0       ACCGGCACGCCCCCC :F,,,F,F,F,FF,F NH:i:7  HI:i:7  AS:i:14 nM:i:0  NM:i:0  MD:Z:15 RG:Z:c949090    SA:Z:chr9,9442085,+,15S111M,0,10;
A00467:240:HCN7KDRX2:2:2275:20410:25786 2465    chr2    113588497       0       1S18M107H       chr14   49586604        0       GGGGGGCGTGCCGGTGGTC     F,FF,F,F,F,,,F:,F:,     NH:i:7  HI:i:5  AS:i:17 nM:i:0  NM:i:0  MD:Z:18 RG:Z:c949090    SA:Z:chr14,49586609,+,19S107M,0,10;
A00467:240:HCN7KDRX2:2:2275:20410:25786 419     chr9    9442085 0       15S111M =       9442086 125     GGGGGGCGTGCCGGTGGTCCCAGCAACACGGGCGGCTGAGGCGGGAGGATCGATGGAGTCAAGGAGTTCTGGGGTGTAGTGCGCTATGCCGATCGGGTGTCCGCACTAACTTCGGCATCAAAATGG  F,FF,F,F,F,,,F:,F:,,::,F,F,,F,::,,::,:,F,:,:::F:,,::,,,:F:::,::::::,:F,FF,:F,:F,F:::FF,:,:::,F,:,,:::FFFFF,,F,,,:,:FFF:FF,F,,,  NH
:i:7    HI:i:7  AS:i:213        nM:i:10 NM:i:10 MD:Z:9T2T4A9T9C1T5C12C35G11T4   RG:Z:c949090    SA:Z:chr1,46341589,-,111H15M,0,0;
A00467:240:HCN7KDRX2:2:2275:20410:25786 339     chr9    9442086 0       2S124M  =       9442085 -125    TAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGGATCGCTTGAGTCCAGGAGTTCTGGGCTGTAGTGCGCTATGCCGATCGGGTGTCCGCACTAAGTTCGGCATCAATATGGTGACCTCCCGGGAG  FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF  NH
:i:7    HI:i:7  AS:i:213        nM:i:10 NM:i:0  MD:Z:124        RG:Z:c949090
A00467:240:HCN7KDRX2:2:2275:20410:25786 2225    chr12   26990   0       107H18M1S       chr14   49586604        0       GACCACCGGCACGCCCCCC     ,:F,:F,,,F,F,F,FF,F     NH:i:7  HI:i:2  AS:i:17 nM:i:0  NM:i:0  MD:Z:18 RG:Z:c949090    SA:Z:chr14,49586609,+,19S107M,0,10;
A00467:240:HCN7KDRX2:2:2275:20410:25786 83      chr14   49586604        0       126M    =       49586609        -121    TAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGGATCGCTTGAGTCCAGGAGTTCTGGGCTGTAGTGCGCTATGCCGATCGGGTGTCCGCACTAAGTTCGGCATCAATATGGTGACCTCCCGGGAG  FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
FFFFFFFFFFFF    NH:i:7  HI:i:2  AS:i:211        nM:i:10 NM:i:0  MD:Z:126        RG:Z:c949090
A00467:240:HCN7KDRX2:2:2275:20410:25786 339     chr14   49586604        0       126M    =       49586606        -124    TAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGGATCGCTTGAGTCCAGGAGTTCTGGGCTGTAGTGCGCTATGCCGATCGGGTGTCCGCACTAAGTTCGGCATCAATATGGTGACCTCCCGGGAG  FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
FFFFFFFFFFFF    NH:i:7  HI:i:1  AS:i:214        nM:i:10 NM:i:0  MD:Z:126        RG:Z:c949090
A00467:240:HCN7KDRX2:2:2275:20410:25786 339     chr14   49586604        0       126M    =       49586609        -121    TAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGGATCGCTTGAGTCCAGGAGTTCTGGGCTGTAGTGCGCTATGCCGATCGGGTGTCCGCACTAAGTTCGGCATCAATATGGTGACCTCCCGGGAG  FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
FFFFFFFFFFFF    NH:i:7  HI:i:3  AS:i:211        nM:i:10 NM:i:0  MD:Z:126        RG:Z:c949090
A00467:240:HCN7KDRX2:2:2275:20410:25786 339     chr14   49586604        0       126M    =       49586609        -121    TAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGGATCGCTTGAGTCCAGGAGTTCTGGGCTGTAGTGCGCTATGCCGATCGGGTGTCCGCACTAAGTTCGGCATCAATATGGTGACCTCCCGGGAG  FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
FFFFFFFFFFFF    NH:i:7  HI:i:4  AS:i:211        nM:i:10 NM:i:0  MD:Z:126        RG:Z:c949090
A00467:240:HCN7KDRX2:2:2275:20410:25786 339     chr14   49586604        0       126M    =       49586609        -121    TAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGGATCGCTTGAGTCCAGGAGTTCTGGGCTGTAGTGCGCTATGCCGATCGGGTGTCCGCACTAAGTTCGGCATCAATATGGTGACCTCCCGGGAG  FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
FFFFFFFFFFFF    NH:i:7  HI:i:5  AS:i:211        nM:i:10 NM:i:0  MD:Z:126        RG:Z:c949090
A00467:240:HCN7KDRX2:2:2275:20410:25786 339     chr14   49586604        0       126M    =       49586609        -121    TAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGGATCGCTTGAGTCCAGGAGTTCTGGGCTGTAGTGCGCTATGCCGATCGGGTGTCCGCACTAAGTTCGGCATCAATATGGTGACCTCCCGGGAG  FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
FFFFFFFFFFFF    NH:i:7  HI:i:6  AS:i:211        nM:i:10 NM:i:0  MD:Z:126        RG:Z:c949090
A00467:240:HCN7KDRX2:2:2275:20410:25786 419     chr14   49586606        0       16S110M =       49586604        124     GGGGGGCGTGCCGGTGGTCCCAGCAACACGGGCGGCTGAGGCGGGAGGATCGATGGAGTCAAGGAGTTCTGGGGTGTAGTGCGCTATGCCGATCGGGTGTCCGCACTAACTTCGGCATCAAAATGG  F,FF,F,F,F,,,F:,F:,,::,F,F,,F,::,,::,:,F,:,:::F:,,::,,,:F:::,::::::,:F,FF,:F,:F,F:::FF,:,:::,F,:,,:::FFFFF,,F,,,:,
:FFF:FF,F,,,    NH:i:7  HI:i:1  AS:i:214        nM:i:10 NM:i:10 MD:Z:8T2T4A9T9C1T5C12C35G11T4   RG:Z:c949090    SA:Z:chr1,46341588,-,110H16M,0,1;
A00467:240:HCN7KDRX2:2:2275:20410:25786 163     chr14   49586609        0       19S107M =       49586604        121     GGGGGGCGTGCCGGTGGTCCCAGCAACACGGGCGGCTGAGGCGGGAGGATCGATGGAGTCAAGGAGTTCTGGGGTGTAGTGCGCTATGCCGATCGGGTGTCCGCACTAACTTCGGCATCAAAATGG  F,FF,F,F,F,,,F:,F:,,::,F,F,,F,::,,::,:,F,:,:::F:,,::,,,:F:::,::::::,:F,FF,:F,:F,F:::FF,:,:::,F,:,,:::FFFFF,,F,,,:,
:FFF:FF,F,,,    NH:i:7  HI:i:2  AS:i:211        nM:i:10 NM:i:10 MD:Z:5T2T4A9T9C1T5C12C35G11T4   RG:Z:c949090    SA:Z:chr12,26990,-,107H18M1S,0,0;
A00467:240:HCN7KDRX2:2:2275:20410:25786 419     chr14   49586609        0       19S107M =       49586604        121     GGGGGGCGTGCCGGTGGTCCCAGCAACACGGGCGGCTGAGGCGGGAGGATCGATGGAGTCAAGGAGTTCTGGGGTGTAGTGCGCTATGCCGATCGGGTGTCCGCACTAACTTCGGCATCAAAATGG  F,FF,F,F,F,,,F:,F:,,::,F,F,,F,::,,::,:,F,:,:::F:,,::,,,:F:::,::::::,:F,FF,:F,:F,F:::FF,:,:::,F,:,,:::FFFFF,,F,,,:,
:FFF:FF,F,,,    NH:i:7  HI:i:3  AS:i:211        nM:i:10 NM:i:10 MD:Z:5T2T4A9T9C1T5C12C35G11T4   RG:Z:c949090    SA:Z:chr1,195452,-,107H18M1S,0,0;
A00467:240:HCN7KDRX2:2:2275:20410:25786 419     chr14   49586609        0       19S107M =       49586604        121     GGGGGGCGTGCCGGTGGTCCCAGCAACACGGGCGGCTGAGGCGGGAGGATCGATGGAGTCAAGGAGTTCTGGGGTGTAGTGCGCTATGCCGATCGGGTGTCCGCACTAACTTCGGCATCAAAATGG  F,FF,F,F,F,,,F:,F:,,::,F,F,,F,::,,::,:,F,:,:::F:,,::,,,:F:::,::::::,:F,FF,:F,:F,F:::FF,:,:::,F,:,,:::FFFFF,,F,,,:,
:FFF:FF,F,,,    NH:i:7  HI:i:4  AS:i:211        nM:i:10 NM:i:10 MD:Z:5T2T4A9T9C1T5C12C35G11T4   RG:Z:c949090    SA:Z:chr1,24927,-,107H18M1S,0,0;
A00467:240:HCN7KDRX2:2:2275:20410:25786 419     chr14   49586609        0       19S107M =       49586604        121     GGGGGGCGTGCCGGTGGTCCCAGCAACACGGGCGGCTGAGGCGGGAGGATCGATGGAGTCAAGGAGTTCTGGGGTGTAGTGCGCTATGCCGATCGGGTGTCCGCACTAACTTCGGCATCAAAATGG  F,FF,F,F,F,,,F:,F:,,::,F,F,,F,::,,::,:,F,:,:::F:,,::,,,:F:::,::::::,:F,FF,:F,:F,F:::FF,:,:::,F,:,,:::FFFFF,,F,,,:,
:FFF:FF,F,,,    NH:i:7  HI:i:5  AS:i:211        nM:i:10 NM:i:10 MD:Z:5T2T4A9T9C1T5C12C35G11T4   RG:Z:c949090    SA:Z:chr2,113588497,+,1S18M107H,0,0;
A00467:240:HCN7KDRX2:2:2275:20410:25786 419     chr14   49586609        0       19S107M =       49586604        121     GGGGGGCGTGCCGGTGGTCCCAGCAACACGGGCGGCTGAGGCGGGAGGATCGATGGAGTCAAGGAGTTCTGGGGTGTAGTGCGCTATGCCGATCGGGTGTCCGCACTAACTTCGGCATCAAAATGG  F,FF,F,F,F,,,F:,F:,,::,F,F,,F,::,,::,:,F,:,:::F:,,::,,,:F:::,::::::,:F,FF,:F,:F,F:::FF,:,:::,F,:,,:::FFFFF,,F,,,:,
:FFF:FF,F,,,    NH:i:7  HI:i:6  AS:i:211        nM:i:10 NM:i:10 MD:Z:5T2T4A9T9C1T5C12C35G11T4   RG:Z:c949090    SA:Z:chr19,66535,-,107H18M1S,0,0;
A00467:240:HCN7KDRX2:2:2275:20410:25786 2481    chr19   66535   0       107H18M1S       chr14   49586604        0       GACCACCGGCACGCCCCCC     ,:F,:F,,,F,F,F,FF,F     NH:i:7  HI:i:6  AS:i:17 nM:i:0  NM:i:0  MD:Z:18 RG:Z:c949090    SA:Z:chr14,49586609,+,19S107M,0,10;
```

This is where Arriba specifies its STAR parameters: https://github.com/suhrig/arriba/blob/786f647a73b8ffb8bbe38b6607460c27e17e8846/run_arriba.sh#L30-L36